File: problems/src/test/java/AddTwoNumbersTest.java
import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class AddTwoNumbersTest extends SampledTest {
    @Override
    public String getFileName() {
        return "add_two_numbers.json";
    }

    @Test
    public void test() {
        for (TypedSample sample : getSamples(TypedSample[].class)) {
            for (AddTwoNumbers solution : AddTwoNumbers.values()) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.addTwoNumbers(
                            SinglyListNode.of(sample.input.l1),
                            SinglyListNode.of(sample.input.l2)
                        )
                    ).isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, int[]> {
        public static class Input {
            public int[] l1;
            public int[] l2;
        }
    }
}

File: typescript/__tests__/problems1_100/add-two-numbers.test.ts
import {deepEqual} from 'assert';
import SinglyListNode from '../../src/concepts/singly-list-node';
import SOLUTIONS from '../../src/problems1_100/add-two-numbers';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    l1: number[],
    l2: number[],
  },
  output: number[],
  message: string,
}

describe(
    'AddTwoNumbers',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach((sample: Sample) =>
                        deepEqual(
                            solution.addTwoNumbers(
                                SinglyListNode.of(...sample.input.l1),
                                SinglyListNode.of(...sample.input.l2),
                            ),
                            SinglyListNode.of(...sample.output),
                            sample.message,
                        ),
                    ),
                ),
        ),
);
